### PR TITLE
NFS: Revamp example, add nfs-common to debian saltbase

### PR DESCRIPTION
**Pull request by zmerlynn**
This ensures nfs-common is installed on GCE, and provides a more
functional explanation/example. I launched two pods so that there was
a busybox pod to poke around at the NFS volume, and so that the later
wget actually works (the original example would have to work on the
node, or need some other access to the container network). After
switching to two pods, it actually makes more sense to use PV claims,
and it's probably a configuration that makes more sense for
indirection for NFS anyways.

### CODE REVIEW

**Comment by zmerlynn** [comment]
cc @thockin @jsafrane

**Comment by zmerlynn** [comment]
@k8s-bot test this

**Comment by zmerlynn** [comment]
See #15995 for an explanation of the hardcoded IP.

File: examples/nfs/nfs-busybox.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#
+# This mounts the nfs volume claim into /mnt
+#
+
+apiVersion: v1
+kind: Pod
```
Comment (bgrant0607) [inline examples/nfs/nfs-busybox.yaml]: Why not a RC?

File: examples/nfs/nfs-web-pod.yaml [hunk 2]
```diff
@@ -1,11 +1,13 @@
 #
-# This pod imports nfs-server.default.kube.local:/ into /usr/share/nginx/html
+# This pod imports the nfs volume claim into /usr/share/nginx/html
 #
 
 apiVersion: v1
 kind: Pod
```
Comment (bgrant0607) [inline examples/nfs/nfs-web-pod.yaml]: Why not a RC?

File: examples/nfs/README.md [hunk 3]
```diff
@@ -33,57 +33,105 @@ Documentation for other releases can be found at
...
 into another one.
 
-### Prerequisites
+### NFS server part
+
+Define [NFS server pod](nfs-server-pod.yaml) and
+[NFS service](nfs-server-service.yaml):
+
+```console
+$ kubectl create -f examples/nfs/nfs-server-pod.yaml
```
Comment (bgrant0607) [inline examples/nfs/README.md]: Why not a RC?

File: examples/nfs/nfs-web-pod.yaml (hunk 2 above)
Comment (zmerlynn) [inline examples/nfs/nfs-web-pod.yaml]: Will convert to RCs while I'm here. It's fast enough.

**Comment by jsafrane** [comment]
I know it's not part of your change, but would you consider using gcr.io/google_containers/volume-nfs as NFS server? It's from test/images/volumes-tester/nfs/ and I hope it will be tested by e2e tests more often than my private jsafrane/nfs-data image.

**Comment by zmerlynn** [comment]
@jsafrane: I absolutely would, but I just spent an hour or so futzing with the official image and while your private image seems to work fine in the example, the official image seems to be broken, which is actually kind of disturbing. I'm going to put this PR up with the private image and then try to iterate on what's going on there, because it *looks* like it should be a drop-in replacement.

**Comment by zmerlynn** [comment]
PTAL, I believe this is ready. I switched everything over to ReplicationControllers. As mentioned above, I attempts to move to `gcr.io/google_containers/volume-nfs` and will continue iterating on that separately, but I think that can stand separately.

**Comment by jsafrane** [comment]
LGTM

**Comment by zmerlynn** [comment]
@jsafrane: I figured out what was going on with the `gcr.io/google_containers/volume-nfs` image and fixed it. I'm not sure how storage e2es would be working, but `insecure` is necessary on my configuration, and was enabled on the `examples` server image but not the test server image. The incoming port number is almost entirely irrelevant (the "security" provided by the privileged port number range for NFS has been laughable for .. forever), and in the container environment it's almost certainly going to be true.

**Comment by jsafrane** [comment]
@zmerlynn: good catch, `insecure` was (wrongly) removed by my commit ebcc558734d92f85c9d50095b83474cb42439141

**Comment by zmerlynn** [comment]
@thockin: Know you're busy, but ping. :)

File: examples/nfs/README.md [hunk 4]
```diff
@@ -33,57 +33,115 @@ Documentation for other releases can be found at
...
+NAME                READY     STATUS    RESTARTS   AGE
+nfs-busybox-jdhf3   1/1       Running   0          25m
+nfs-busybox-w3s4t   1/1       Running   0          25m
+$ kubectl exec nfs-busybox-jdhf3 -- cat /mnt/index.html
+Thu Oct 22 19:20:18 UTC 2015
+nfs-busybox-w3s4t
+```
+
+You should see output similar to the above if everything is working well. If
+it's not, make sure you changed the invalid IP in the [nfs PV](nfs-pv.yaml) file
```
Comment (thockin) [inline examples/nfs/README.md]: s/nfs/NFS

File: examples/nfs/README.md [hunk 5]
```diff
@@ -33,57 +33,115 @@ Documentation for other releases can be found at
...
+volume and runs a simple web server on it.
 
 Define the pod:
 
-    $ kubectl create -f nfs-web-pod.yaml
+```console
+$ kubectl create -f examples/nfs/nfs-web-rc.yaml
+```
+
+This creates two pods, each of which server the `index.html` from above. We can
```
Comment (thockin) [inline examples/nfs/README.md]: s/server/serve

File: test/images/volumes-tester/nfs/Makefile [hunk 6]
```diff
@@ -1,6 +1,6 @@
 all: push
 
-TAG = 0.3
+TAG = 0.4
```
Comment (thockin) [inline test/images/volumes-tester/nfs/Makefile]: already pushed?

**Comment by thockin** [comment]
nits: feel free to self-lgtm

File: test/images/volumes-tester/nfs/Makefile (hunk 6 above)
Comment (zmerlynn) [inline test/images/volumes-tester/nfs/Makefile]: Nope, was waiting for LGTM (will push now).

**Comment by zmerlynn** [comment]
Nits addressed. Thanks for the review!

**Comment by zmerlynn** [comment]
@k8s-bot unit test this please

**Comment by zmerlynn** [comment]
@k8s-bot unit test this please

**Comment by zmerlynn** [comment]
@k8s-bot unit test this please

**Pull request merged**